salt/modules/vsphere.py

```diff
@@ -3519,3 +3532,140 @@ def _set_syslog_config_helper(host, username, password, syslog_config, config_va
+
+
+def add_host_to_dvs(host, username, password, vmknic_name, vmnic_name,
+                    dvs_name, portgroup_name, protocol=None, port=None,
+                    host_names=None):
+    '''
+    Adds an ESXi host to a vSphere Distributed Virtual Switch
+    DOES NOT migrate the ESXi's physical and virtual NICs to the switch (yet)
+    (please don't remove the commented code)
+    '''
+    service_instance = salt.utils.vmware.get_service_instance(host=host,
+                                                              username=username,
+                                                              password=password,
+                                                              protocol=protocol,
+                                                              port=port)
+    dvs = salt.utils.vmware._get_dvs(service_instance, dvs_name)
+    target_portgroup = salt.utils.vmware._get_dvs_portgroup(dvs,
+                                                            portgroup_name)
+    uplink_portgroup = salt.utils.vmware._get_dvs_uplink_portgroup(dvs,
+                                                                   'DSwitch-DVUplinks-34')
+    dvs_uuid = dvs.config.uuid
+    host_names = _check_hosts(service_instance, host, host_names)
+
+    ret = {}
+    for host_name in host_names:
+        # try:
+        ret[host_name] = {}
+
+        ret[host_name].update({'status': False,
+                               'portgroup': portgroup_name,
+                               'dvs': dvs_name})
+        host_ref = _get_host_ref(service_instance, host, host_name)
+
+        dvs_hostmember_config = vim.dvs.HostMember.ConfigInfo(
+            host=host_ref
+        )
+        dvs_hostmember = vim.dvs.HostMember(
+            config=dvs_hostmember_config
+        )
+        p_nics = salt.utils.vmware._get_pnics(host_ref)
+        p_nic = [x for x in p_nics if x.device == 'vmnic0']
+        v_nics = salt.utils.vmware._get_vnics(host_ref)
+        v_nic = [x for x in v_nics if x.device == vmknic_name]
+        v_nic_mgr = salt.utils.vmware._get_vnic_manager(host_ref)
+
+        dvs_pnic_spec = vim.dvs.HostMember.PnicSpec(
+            pnicDevice=vmnic_name,
+            uplinkPortgroupKey=uplink_portgroup.key
+        )
+        pnic_backing = vim.dvs.HostMember.PnicBacking(
+            pnicSpec=[dvs_pnic_spec]
+        )
+        dvs_hostmember_config_spec = vim.dvs.HostMember.ConfigSpec(
+            host=host_ref,
+            operation='add',
+        )
+        dvs_config = vim.DVSConfigSpec(
+                configVersion=dvs.config.configVersion,
+                host=[dvs_hostmember_config_spec])
+        task = dvs.ReconfigureDvs_Task(spec=dvs_config)
+        salt.utils.vmware.wait_for_task(task, host_name,
+                                        'Adding host to the DVS',
+                                        sleep_seconds=3)
+
+        ret[host_name].update({'status': True})
```

Comment:
This function doesn't appear to return anything. Probably just needs a `return ret`.